src/Core/Content/Seo/SalesChannel/StoreApiSeoResolver.php

```diff
@@ -60,14 +60,24 @@ public function addSeoInformation(ResponseEvent $event): void
             return;
         }
 
-        if (!$event->getRequest()->headers->has(PlatformRequest::HEADER_INCLUDE_SEO_URLS)) {
+        $request = $event->getRequest();
+
+        if (!$request->headers->has(PlatformRequest::HEADER_INCLUDE_SEO_URLS)) {
+            return;
+        }
+
+        /** @var SalesChannelContext|null $context */
+        $context = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_CONTEXT_OBJECT);
+
+        // This is likely the case for routes with the `auth_required` option set to `false`.
+        if (null === $context) {
```

Comment:
Can you add a testcase for this case in `\Shopware\Core\Content\Test\Seo\SalesChannel\StoreApiSeoResolverTest` ?

Reply:
How would you suggest to test it there? I've thought about this too, but didn't get anywhere yet.

Testing it properly requires a route with `auth_required` set to `false` and with either products or categories in the response. As of now there doesn't seem to be such a route.

I found `\Shopware\Tests\Integration\Core\Framework\Api\EventListener\FixturesPhp\SalesChannelAuthenticationListenerTestRoute`, but I'm not sure if it'd be ok to use it in the test.